src/blackbox_decode.c

```diff
@@ -454,11 +454,21 @@ void outputGPSFields(flightLog_t *log, FILE *file, int64_t *frame)
     }
 }
 
+int getMajorVersion(flightLog_t *log) {
+	if (!log->private->fcVersion[0]) return -1; //Unknown version
+	char fcVersion[30];
+	strcpy(fcVersion, log->private->fcVersion);
+	return atoi(strtok(fcVersion,"."));
+}
+
 /**
- * Get altitude in [m] from betaflight logged [cm], including optional user altitude offset.
+ * Get altitude in [m] including optional user altitude offset.
+ * Note: Betaflight before version 4 logged altitude in [cm], since Betaflight 4 it use [dm].
  */
 float getAltitude(flightLog_t *log, int64_t *frame) {
-	return frame[log->gpsFieldIndexes.GPS_altitude] / 100.0 + options.altOffset; //Change [cm] to [m] for gpx format
+	int majorFcVersion=getMajorVersion(log);
```

Comment:
Please use spaces around operators.

Reply:
done

Reply:
The spaces around `=` are still missing.